faust/sensors/monitor.py

```diff
@@ -222,6 +242,24 @@ def _topic_buffer_full_dict(self) -> MutableMapping[str, int]:
     def _metric_counts_dict(self) -> MutableMapping[str, int]:
         return {key: count for key, count in self.metric_counts.items()}
 
+    def _tp_committed_offsets_dict(self) -> TPOffsetDict:
+        return self._tp_offsets_as_dict(self.tp_committed_offsets)
+
+    def _tp_read_offsets_dict(self) -> TPOffsetDict:
+        return self._tp_offsets_as_dict(self.tp_read_offsets)
+
+    def _tp_end_offsets_dict(self) -> TPOffsetDict:
+        return self._tp_offsets_as_dict(self.tp_end_offsets)
+
+    @classmethod
+    def _tp_offsets_as_dict(cls, tp_offsets: TPOffsetMapping) -> TPOffsetDict:
+        topic_partition_offsets: TPOffsetDict = {}
+        for tp, offset in tp_offsets.items():
+            partition_offsets = topic_partition_offsets.get(tp.topic, {})
```

Comment:
nit: `topic_partition_offsets.get(tp.topic) or {}`

it's better because it only creates the dict when the key does not exist or is None.

Reply:
Ah ok!